pages/index.js

```diff
@@ -2,81 +2,35 @@ import React from "react";
-  var tdt = new Date(dt.valueOf());
-  var dayn = (dt.getDay() + 6) % 7;
-  tdt.setDate(tdt.getDate() - dayn + 3);
-  var firstThursday = tdt.valueOf();
-  tdt.setMonth(0, 1);
-  if (tdt.getDay() !== 4) {
-    tdt.setMonth(0, 1 + ((4 - tdt.getDay() + 7) % 7));
-  }
-  return 1 + Math.ceil((firstThursday - tdt) / 604800000);
-}
+import NewsCard from "../src/components/NewsCard";
+import TeamsList from "../src/components/TeamsList";
 
 export default function Home({ data }) {
   const { user, loading } = useFetchUser();
   return (
     <Layout user={user} loading={loading}>
       {loading && <p>Chargement....</p>}
 
-      {!loading &&
-        data &&
-        data.map &&
-        data.map(d => (
-          <div className="card" key={d.file.name}>
-            <div className="card-header">
-              <span>Semaine {getWeekNumber(d.content.date)}</span>
-              <span> - </span>
-              <span>{d.content.mood}</span>
-              <span>{d.content.team}</span>
-            </div>
-            <ul className="list-group list-group-flush">
-              <li className="list-group-item">
-                <dl className="row">
-                  <dt className="col-sm-3">Priorités:</dt>
-                  <dd className="col-sm-9">{d.content.priorities}</dd>
-                </dl>
-              </li>
-              <li className="list-group-item">
-                <dl className="row">
-                  <dt className="col-sm-3">KPIs / OKR:</dt>
-                  <dd className="col-sm-9">{d.content.kpis}</dd>
-                </dl>
-              </li>
-              <li className="list-group-item">
-                <dl className="row">
-                  <dt className="col-sm-3">Échéances:</dt>
-                  <dd className="col-sm-9">{d.content.needs}</dd>
-                </dl>
-              </li>
-              <li className="list-group-item">
-                <dl className="row">
-                  <dt className="col-sm-3">Besoins:</dt>
-                  <dd className="col-sm-9">{d.content.term}</dd>
-                </dl>
-              </li>
-            </ul>
-          </div>
-        ))}
-      <style jsx>{`
-        .card {
-          margin-bottom: 40px;
-        }
-      `}</style>
+      {!loading && data && data.map && (
+        <div className="row">
+          <dd className="col-sm-3">
+            <h4>Produits</h4>
+            <TeamsList />
+          </dd>
+          <dd className="col-sm-9">
+            <h4>Nouvelles</h4>
+            {data.map(d => (
+              <NewsCard data={d} key={d.content.team} />
+            ))}
+          </dd>
+        </div>
+      )}
     </Layout>
   );
 }
 
 Home.getInitialProps = async ({ req }) => {
-  // fetch data for given month
-  const now = new Date();
-  const { year = now.getFullYear(), month = now.getMonth() + 1 } =
-    (req && req.query) || {};
-
-  let url = `/api/fetchPosts?year=${year}&month=${month}`;
+  let url = `/api/fetchLatestPosts`;
```

Comment:
i guess we could directly call the method from the api route here so we save one internal request

Reply:
Alright. I must refactor the 2 calls fetching the published posts anyway :)

Reply:
I've open a specific issue for this one: [issue 13](https://github.com/SocialGouv/carnets/issues/13)